src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Discovery/IPluginsDiscoveryManager.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Performance.Toolkit.Plugins.Core;
+using Microsoft.Performance.Toolkit.Plugins.Core.Discovery;
+using Microsoft.Performance.Toolkit.Plugins.Runtime.Events;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Discovery
+{
+    /// <summary>
+    ///     Contains logic for discovering plugins.
+    /// </summary>
+    public interface IPluginsDiscoveryManager
+    {
+        /// <summary>
+        ///     Gets all plugin sources that this plugin manager is configured to discover plugins from.
+        /// </summary>
+        IEnumerable<PluginSource> PluginSources { get; }
```

Comment:
I would remove this property and make this interface extend `IRepository<PluginSource>`